src/modules/EventModule.cpp

```diff
@@ -0,0 +1,792 @@
+								embed.setDescription("**RAFFLE DRAW** event has **ended**!\n"
+									"**The lucky ticket which wins the draw is `" + tiketId + "`**\n"
+									"**The owner can claim their rewards within 24 hours**");
+								config.claimedReward = false;
+								config.currentTicketIndex = 0;
+
+								//Get the user who has the lucky ticket
+								for (auto& user : raffleDrawGuildList[guildID])
+								{
+									for (auto& userTicket : user.m_TicketIds)
+									{
+										if (luckyTicketId == userTicket)
+										{
+											config.luckyUser = user.m_UserId;
+										}
+									}
+								}
+
+								//Clear all the tickets
+								for (auto& user : raffleDrawGuildList[guildID])
+									user.m_TicketIds.clear();
+
+								//Set the reward claiming time (24 hours)
+								config.raffleDrawRewardClaimTimer = new QTimer;
+								config.raffleDrawRewardClaimTimer->setInterval(24 * 3600000);
+								config.raffleDrawRewardClaimTimer->setSingleShot(true);
+
+								QObject::connect(config.raffleDrawRewardClaimTimer, &QTimer::timeout, [this, guildID, chan, authorId]()
+									{
+										auto& config = getServerEventData(guildID);
+										config.claimedReward = true;
+										UmikoBot::Instance().createMessage(chan, "The owner of the price in the RaffleDraw didn't get their rewards!\n"
+																				"The role(s) who have permission can start this event again!");
+									});
+								config.raffleDrawRewardClaimTimer->start();
+								config.isEventRunning = false;
+								config.eventRaffleDrawRunning = false;
+								UmikoBot::Instance().createMessage(chan, embed);
+							}
+						});
+					config.eventTimer->start();
+				});
+	});
+	RegisterCommand(Commands::EVENT_END, "endevent", [this](Client& client, const Message& message, const Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+
+		client.getGuildMember(channel.guildId(), message.author().id())
+			.then([=](const GuildMember& member)
+				{
+					bool found = false;
+
+					for (auto& allowedRole : m_EventWhitelist[channel.guildId()])
+					{
+						if (member.roles().contains(allowedRole))
+						{
+							found = true;
+							break;
+						}
+					}
+
+					if (!found)
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**You don't have permissions to end events.**");
+						return;
+					}
+					auto& config = getServerEventData(channel.guildId());
+					CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+					auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+
+					QString eventEmoji = QString(utility::consts::emojis::REGIONAL_INDICATOR_E) + " " + QString(utility::consts::emojis::REGIONAL_INDICATOR_V) + " " +
+											QString(utility::consts::emojis::REGIONAL_INDICATOR_E) + " " + QString(utility::consts::emojis::REGIONAL_INDICATOR_N) + " " +
+												QString(utility::consts::emojis::REGIONAL_INDICATOR_T);
```

Comment:
You don't need to hype up the end of an event.

Reply:
It needs to attract the eyes, if I remove the current title of the embed, Please give me a better title and a description. Yes, the description is needed I think because, the only title wouldn't attract the users attention, its so small.